src/precompiles/native.rs

```diff
@@ -45,95 +44,149 @@ impl ExitToNear {
         super::make_address(0xe9217bc7, 0x0b7ed1f598ddd3199e80b093fa71124f);
 }
 
-impl Precompile for ExitToNear {
+fn get_nep141_from_erc20(erc20_token: &[u8]) -> AccountId {
+    AccountId::from_utf8(
+        crate::sdk::read_storage(bytes_to_key(KeyPrefix::Erc20Nep141Map, erc20_token).as_slice())
+            .expect(ERR_TARGET_TOKEN_NOT_FOUND),
+    )
+    .unwrap()
```

Comment:
Absolutely, actually I think we should find a way to totally disallow calling `crate::sdk::read_storage` at all (by making this function non-public) and only available for `storage.rs` module. Otherwise building abstraction and doing migrations will be crazy.

Reply:
Well, after a second thought, this is tricky here, because inside the precompile we don't have access to the Engine internal state. We can recreate it by calling Engine::new() and then using proper abstraction, but that would be a waste of resources loading the engine state again.

That makes me think, that I'll change implementation a bit. Instead of storing bijection map as a field, I'll have static function that builds the interface for the bijection map (no state need to be touched for this operation), here it is very helpful to have it implemented with keys as generic consts.

Reply:
> Instead of storing bijection map as a field, I'll have static function that builds the interface for the bijection map

Yeah, this is a cool idea. I like it better than having the raw lookup here. I'm a little worried about whether this will be an abuse of the environment (it relies on the existence of ambient storage) which might be hard to reproduce in unit tests. But maybe that is a necessary evil and these functions will only ever be covered by integration tests.

Reply:
Implemented.